src/main/webapp/app/shared/exercise-filter/exercise-filter-modal.helper.ts

```diff
@@ -0,0 +1,77 @@
+import { SidebarCardElement } from 'app/types/sidebar';
+import { DifficultyLevel, ExerciseType } from 'app/entities/exercise.model';
+import { ExerciseCategory } from 'app/entities/exercise-category.model';
+import { FilterDetails, RangeFilter } from 'app/types/exercise-filter';
+import { getLatestResultOfStudentParticipation } from 'app/exercises/shared/participation/participation.utils';
+
+export function satisfiesDifficultyFilter(sidebarElement: SidebarCardElement, searchedDifficulties?: DifficultyLevel[]): boolean {
+    if (!searchedDifficulties?.length) {
+        return true;
+    }
+    if (!sidebarElement.difficulty) {
+        return false;
+    }
+
+    return searchedDifficulties.includes(sidebarElement.difficulty);
+}
+
+export function satisfiesTypeFilter(sidebarElement: SidebarCardElement, searchedTypes?: ExerciseType[]): boolean {
+    if (!searchedTypes?.length) {
+        return true;
+    }
+    if (!sidebarElement.exercise?.type) {
+        return false;
+    }
+
+    return searchedTypes.includes(sidebarElement.exercise.type);
+}
+
+export function satisfiesCategoryFilter(sidebarElement: SidebarCardElement, selectedCategories: ExerciseCategory[]): boolean {
+    if (!selectedCategories.length) {
+        return true;
+    }
+    if (!sidebarElement?.exercise?.categories) {
+        return false;
+    }
+
+    return sidebarElement.exercise.categories.some((category) => selectedCategories.some((selectedCategory) => selectedCategory.equals(category)));
+}
+
+export function satisfiesScoreFilter(sidebarElement: SidebarCardElement, isFilterApplied: boolean, achievedScoreFilter?: RangeFilter): boolean {
+    if (!isFilterApplied || !achievedScoreFilter) {
+        return true;
+    }
+    if (!sidebarElement.studentParticipation) {
+        return achievedScoreFilter.filter.selectedMin === 0;
+    }
+
+    const latestResult = getLatestResultOfStudentParticipation(sidebarElement.studentParticipation, true);
+    if (!latestResult?.score) {
+        return achievedScoreFilter.filter.selectedMin === 0;
+    }
```

Comment:
`getLatestResultOfStudentParticipation` expects a `Participation` and does not accept `undefined`

I was hesitant to change it because this util function is not (directly) tested. However, allowing `undefined` and adding the check within `getLatestResultOfStudentParticipation` should not be able to break anything as passing this value was not allowed beforehand anyways.

I moved the undefined check to `getLatestResultOfStudentParticipation`